referenceframe/frame_system.go

```diff
@@ -397,6 +401,44 @@ func (sfs *simpleFrameSystem) getFrameToWorldTransform(inputMap map[string][]Inp
 	return srcToWorld, err
 }
 
+// ReplaceFrame transfers replaceMe's children and parentage to replaceWith. ReplaceMe is removed entirely from the frame system.
+// ReplaceWith is not allowed to exist within the frame system at the time of the call.
+func (sfs *simpleFrameSystem) ReplaceFrame(fs FrameSystem, replaceMe, replaceWith Frame) error {
```

Comment:
Note for @biotinker : I think its ok to write this function since we have established a precedence for FSs being mutable, but curious to get your thoughts on it

Reply:
Yeah I think this is okay. The FS service will return a new FS object at each `FrameSystem()` call, so this shouldn't affect anything else.